### PR TITLE
feat(schedule): Log descriptive cron schedules

### PR DESCRIPTION
**Pull request by Sukhdev2002**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs tests, including if an invalid cron string is passed

**Comment by Sukhdev2002** [comment]
> Needs tests, including if an invalid cron string is passed

test case already written for invalid string types

**Comment by rarkins** [comment]
You had tests before but they're missing now

**Comment by Sukhdev2002** [comment]
However, all the scenarios have already been covered in the previous test cases. Additionally, the test cases I added earlier will not run for the function where I made the changes.

**Comment by rarkins** [comment]
We don't add features without tests to verify that the feature works

**Comment by Sukhdev2002** [comment]
> We don't add features without tests to verify that the feature works

added please check

File: lib/workers/repository/update/branch/schedule.spec.ts [hunk 1]
```diff
@@ -407,4 +408,21 @@ describe('workers/repository/update/branch/schedule', () => {
       });
     });
   });
+
+  describe('cronstrue', () => {
```
Comment (rarkins) [inline lib/workers/repository/update/branch/schedule.spec.ts:412]: Needs a test for an invalid value

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/update/branch/schedule.spec.ts (hunk 1 above)
Comment (Sukhdev2002) [reply lib/workers/repository/update/branch/schedule.spec.ts:412]: ok

Comment (Sukhdev2002) [reply lib/workers/repository/update/branch/schedule.spec.ts:412]: > Needs a test for an invalid value

Invalid test case is added

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/update/branch/schedule.spec.ts [hunk 2]
```diff
@@ -407,4 +408,28 @@ describe('workers/repository/update/branch/schedule', () => {
...
+    });
+
+    it('should correctly convert "0 23 * * *" to human-readable format', () => {
+      const result = cronstrue.toString('0 23 * * *');
+      expect(result).toBe('At 11:00 PM');
+    });
+    
+    it('should throw an error for an invalid cron expression "* * */2 6#1"', () => {
+      expect(() => {
+        cronstrue.toString("* * */2 6#1");
```
[the remark this answers is not in the document]
Comment (Sukhdev2002) [reply lib/workers/repository/update/branch/schedule.spec.ts]: > Use throwExceptionOnParseError=false like in the actual code

added

**Review by rarkins — APPROVED** [review]

File: lib/workers/repository/update/branch/schedule.ts [hunk 3]
```diff
@@ -25,6 +26,10 @@ function parseCron(
   timezone?: string,
 ): CronExpression | undefined {
   try {
+    const cronScheduleSummary = cronstrue.toString(scheduleText, {
+      throwExceptionOnParseError: false,
+    });
+    logger.debug(`Human-readable summary for cron:: ${cronScheduleSummary}`);
```
Comment (viceice) [inline lib/workers/repository/update/branch/schedule.ts:32]: this causes a weird debug message for all later based strings as we try to parse as cron first 😕

Comment (viceice) [reply lib/workers/repository/update/branch/schedule.ts:32]: @rarkins we should refactor to only log if it's a cron expression and not a later expression.

Comment (viceice) [reply lib/workers/repository/update/branch/schedule.ts:32]: this message is also logged multiple times (for each branch)

**Pull request merged**